core/state/statedb_utils.go

```diff
@@ -32,11 +36,41 @@ var (
 	}
 )
 
+// IsWhitelistedDeployer reads the contract storage to check if an address is allow to deploy
+func IsWhitelistedDeployerV2(statedb *StateDB, address common.Address, blockTime uint64) bool {
+	contract := common.HexToAddress(common.WhitelistDeployerSCV2)
+	whitelistAllSlot := slotWhitelistDeployerMappingV2[WHITELIST_ALL]
+	whitelistAll := statedb.GetState(contract, GetLocSimpleVariable(whitelistAllSlot))
+
+	if whitelistAll.Big().Cmp(big.NewInt(1)) == 0 {
+		return true
+	}
+
+	whitelistedSlot := slotWhitelistDeployerMappingV2[WHITELISTED]
+	// whiteListed Have 2 fields, so we need to plus 1
+	expiredLoc := GetLocMappingAtKey(address.Hash(), whitelistedSlot)
+	activatedLoc := common.BigToHash(expiredLoc.Big().Add(expiredLoc.Big(), common.Big1))
+	expiredHash := statedb.GetState(contract, expiredLoc)
+
+	activatedHash := statedb.GetState(contract, activatedLoc)
+
+	// (whiteListInfo.activated && block.timestamp < whiteListInfo.expiryTimestamp)
+	// Compare expiredHash  with Blockheader Timestamp
+	if activatedHash.Big().Cmp(big.NewInt(1)) == 0 {
+		if expiredHash.Big().Cmp(new(big.Int).SetUint64(blockTime)) > 0 {
```

Comment:
I look up and do some experiment. Interestingly, as block time is cannot overflow int64, we can use `big.NewInt(int64(blockTime))`, this function can be inlined and save 1 allocation than the `new(big.Int).SetUint64(blockTime)`
Ref: https://github.com/golang/go/issues/29951, https://go-review.googlesource.com/c/go/+/411254